x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/SecurityRestFilter.java

```diff
@@ -108,7 +108,7 @@ public void handleRequest(RestRequest request, RestChannel channel, NodeClient c
 
         final RestRequest wrappedRequest = maybeWrapRestRequest(request);
         RemoteHostHeader.process(request, threadContext);
-        authenticationService.authenticate(wrappedRequest, ActionListener.wrap(authentication -> {
+        authenticationService.authenticate(wrappedRequest.getHttpRequest(), ActionListener.wrap(authentication -> {
```

Comment:
This is just temporary: the `authenticationService.authenticate` method is going to be invoked before a `RestRequest` is built. This is to be handled in the follow-up PR. For now, just call the authenticate method with the HTTP request that's used to construct the `RestRequest`.